include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -3038,6 +3061,86 @@ struct OrtApi {
    */
   ORT_API2_STATUS(HasValue, _In_ const OrtValue* value, _Out_ int* out);
 
+  /** \brief Set custom thread creation function
+  *
+  * \param[in] session options
+  * \param[in] custom thread creation function
+  * 
+  * * \snippet{doc} snippets.dox OrtStatus Return Value
+  */
+  ORT_API2_STATUS(SessionOptionsSetCustomCreateThreadFn, _Inout_ OrtSessionOptions* options, _In_ CustomCreateThreadFn custom_create_thread_fn);
+
+  /** \brief Set creation options for custom thread 
+  *
+  * \param[in] session options
+  * \param[in] custom thread creation options (can be nullptr)
+  * 
+  * * \snippet{doc} snippets.dox OrtStatus Return Value
+  */
+  ORT_API2_STATUS(SessionOptionsSetCustomThreadCreationOptions, _Inout_ OrtSessionOptions* options, _In_ void* custom_thread_creation_options);
+
+  /** \brief Set custom thread join function
+  *
+  * \param[in] session options
+  * \param[in] custom join thread function, must not be nullptr if custom_create_thread_fn is set
+  * 
+  * * \snippet{doc} snippets.dox OrtStatus Return Value
+  */
+  ORT_API2_STATUS(SessionOptionsSetCustomJoinThreadFn, _Inout_ OrtSessionOptions* options, _In_ CustomJoinThreadFn join_custom_thread_fn);
```

Comment:
nit: parameter name for consistency: custom_join_thread_fn

Reply:
Done.